src/cpu/base.hh

```diff
@@ -178,6 +178,10 @@ class BaseCPU : public ClockedObject
      */
     virtual Port &getDataPort() = 0;
 
+    void scheduleSimpoint(
+      Counter end_point,
```

Comment:
The end_point is an instruction number. When user pass in an end_point, it will schedule a SIMPOINT_END exit event at the end_point+current local instructions.